internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/public.go

```diff
@@ -59,6 +63,54 @@ func (h *Handler) createPublicLinkShare(w http.ResponseWriter, r *http.Request,
 		}
 	}
 
+	// check reserved names
+	linkname := r.FormValue("name")
+	quick, _ := strconv.ParseBool(r.FormValue("quicklink")) // no need to check the error - defaults to zero value!
+	if !quick && linkname == QuicklinkName {
+		return nil, &ocsError{
+			Code:    response.MetaBadRequest.StatusCode,
+			Message: fmt.Sprintf("not allowed to use `%s` as name", linkname),
+		}
+
+	}
+
+	// check if a quicklink should be created
+	if quick {
+		_, f, err := h.addFilters(w, r, &provider.Reference{ResourceId: statInfo.Id})
```

Comment:
Since you already have the resource id and you are only interested in the public link filters, you can just do this: 
```golang
linkFilters := []*link.ListPublicSharesRequest_Filter{publicshare.ResourceIDFilter(info.Id)}
```

Then you don't do an extra stat request in `addFilters`